src/plugins/data/public/types.ts

```diff
@@ -57,20 +60,61 @@ export interface DataPublicPluginSetup {
   __enhance: (enhancements: DataPublicPluginEnhancements) => void;
 }
 
+/**
+ * data plugin prewired UI components
+ */
+export interface DataPublicPluginStartUi {
+  IndexPatternSelect: React.ComponentType<IndexPatternSelectProps>;
+  SearchBar: React.ComponentType<StatefulSearchBarProps>;
+}
+
+/**
+ * utilities to generate filters from action context
+ */
+export interface DataPublicPluginStartActions {
+  createFiltersFromValueClickAction: typeof createFiltersFromValueClickAction;
+  createFiltersFromRangeSelectAction: typeof createFiltersFromRangeSelectAction;
+}
+
+/**
+ * Data plugin public Start contract
+ */
 export interface DataPublicPluginStart {
-  actions: {
-    createFiltersFromValueClickAction: typeof createFiltersFromValueClickAction;
-    createFiltersFromRangeSelectAction: typeof createFiltersFromRangeSelectAction;
-  };
+  /**
+   * filter creation utilities
+   * {@link DataPublicPluginStartActions}
+   */
+  actions: DataPublicPluginStartActions;
+  /**
+   * autocomplete service
+   * {@link AutocompleteStart}
+   */
   autocomplete: AutocompleteStart;
+  /**
+   * index patterns service
+   * {@link IndexPatternsContract}
+   */
   indexPatterns: IndexPatternsContract;
+  /**
+   * search service
+   * {@link ISearchStart}
+   */
   search: ISearchStart;
+  /**
+   * field formats service
+   * {@link FieldFormatsStart}
+   */
   fieldFormats: FieldFormatsStart;
+  /**
+   * query service
+   * {@link QueryStart}
+   */
   query: QueryStart;
-  ui: {
-    IndexPatternSelect: React.ComponentType<IndexPatternSelectProps>;
-    SearchBar: React.ComponentType<StatefulSearchBarProps>;
-  };
+  /**
+   * prewired ui components
```

Comment:
nit, for consistency with other comments

```suggestion
   * prewired UI components
```